docs/resources/custom_physics.md

```
@@ -181,14 +181,28 @@ class CoreBallLCushionCollision(ABC):
 
+For the `dim` requirement, add a `dim` field declaring whether your model is safe in 2D, 3D, or both:
+
+```python
+from pooltool.physics.dimensionality import Dim
+
+dim: Dim = attrs.field(default=Dim.TWO, init=False, repr=False)
+```
+
+`Dim` is a capability declaration consumed at engine construction:
+
+- `Dim.TWO` — your model is safe only when [](#pooltool.evolution.engine.SimulationEngine)'s `is_3d` is `False`. Use this if your model assumes balls are on the table surface (z=R, vz=0).
```

markdownlint-cli2 (0.22.1)

[warning] Link fragments should be valid

(MD051, link-fragments)